src/ripple/app/paths/PathState.h

```diff
@@ -35,20 +36,20 @@ class PathState : public CountedObject <PathState>
     using Ptr = std::shared_ptr<PathState>;
     using List = std::vector<Ptr>;
 
-    PathState (STAmount const& saSend, STAmount const& saSendMax)
+    PathState (PaymentView& parent,
+            STAmount const& saSend,
+                STAmount const& saSendMax)
         : mIndex (0)
         , uQuality (0)
         , saInReq (saSendMax)
         , saOutReq (saSend)
     {
+        view_.emplace(parent);
```

Comment:
Because I would need to use `boost::in_place_factory` and that gets ugly. There's no `boost::optional` constructor that forwards a parameter pack to the contained type.